callback_plugins/failure_log.py

```diff
@@ -0,0 +1,58 @@
+"""
+This callback plugin writes ansible failures to a log as yaml. This way you
+can parse the file later and use the ansible failures for other reporting
+or logging.
+
+A log will not be written unless the environment variable ANSIBLE_FAILURE_LOG
+is present and contains a path to a file to write the log to.
+
+NOTE: This plugin was only tested on ansible version 1.9.2 and does not support
+v2 style plugins
+"""
+import yaml
+import os
+import logging
+
+log = logging.getLogger(__name__)
+# We only want to log if this env var is populated with
+# a file path of where the log should live.
+fail_log = os.environ.get('ANSIBLE_FAILURE_LOG')
+if fail_log:
+    handler = logging.FileHandler(filename=fail_log)
+    log.addHandler(handler)
+
+
+def log_failure(host, result):
+    """
+    If the environment variable ANSIBLE_FAILURE_LOG is present
+    a log of all failures in the playbook will be persisted to
+    the file path given in ANSIBLE_FAILURE_LOG.
+    """
+    failure = {"{0}".format(host): dict()}
+    failure[host] = result
```

Comment:
They could be, yeah.  I don't think it would affect the functionality at all though.